tests/test_modeling_common.py

```diff
@@ -4829,6 +4831,73 @@ def test_forward_with_num_logits_to_keep(self):
+
+        def recursively_check(eager_outputs, exported_outputs):
+            is_tested = False
+            if isinstance(eager_outputs, torch.Tensor):
+                self.assertEqual(eager_outputs.shape, exported_outputs.shape)
+                self.assertTrue(torch.allclose(eager_outputs, exported_outputs, atol=1e-4))
+                return True
+            elif isinstance(eager_outputs, (tuple, list)):
+                for eager_output, exported_output in zip(eager_outputs, exported_outputs):
+                    is_tested = is_tested or recursively_check(eager_output, exported_output)
+                return is_tested
+            return is_tested
+
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        inputs_dict["return_dict"] = False
+
+        # override config kwargs for torch export if needed
+        config_kwargs = config_kwargs if config_kwargs is not None else {}
+        for key, value in config_kwargs.items():
+            setattr(config, key, value)
+
+        for model_class in self.all_model_classes:
+            if model_class.__name__.endswith("ForPreTraining"):
+                continue
+
+            with self.subTest(model_class.__name__):
+                model = model_class(config).eval().to(torch_device)
+
+                # Prepare exported program inputs based on model.forward function signature
+                signature = inspect.signature(model.forward)
+                kwarg_names = [
+                    name
+                    for name, param in signature.parameters.items()
+                    if param.default is not inspect.Parameter.empty
+                ]
+                arg_names = [name for name in signature.parameters.keys() if name not in kwarg_names]
+                inputs_args = tuple(inputs_dict[name] for name in arg_names if name in inputs_dict)
+                inputs_kwargs = {name: inputs_dict[name] for name in kwarg_names if name in inputs_dict}
+
+                # Export model
+                exported_model = torch.export.export(
+                    model,
+                    args=inputs_args,
+                    kwargs=inputs_kwargs,
+                    strict=True,
+                )
+
+                # Test save-restore
+                with tempfile.TemporaryDirectory() as tmpdirname:
+                    save_path = os.path.join(tmpdirname, "exported_model.pt2")
+                    torch.export.save(exported_model, save_path)
+                    exported_model = torch.export.load(save_path)
+
+                # Run exported model and eager model
+                with torch.no_grad():
+                    eager_outputs = model(**inputs_dict)
+                    exported_outputs = exported_model.module().forward(*inputs_args, **inputs_kwargs)
+
+                # Check if outputs are close
+                is_tested = recursively_check(eager_outputs, exported_outputs)
```

Comment:
- `recursively_check` will fail directly if outputs do not match
- `is_tested` shows if we have at least on pair of outputs compared with `allclose`, otherwise we raise an error